engine-plugins/frame-plugins/src/main/scala/org/trustedanalytics/atk/engine/frame/plugins/load/JdbcPlugin/LoadJdbcImpl.scala

```diff
@@ -0,0 +1,73 @@
+/*
+// Copyright (c) 2015 Intel Corporation 
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+*/
+
+package org.trustedanalytics.atk.engine.frame.plugins.load.JdbcPlugin
+
+import org.apache.spark._
+import org.apache.spark.frame.FrameRdd
+import org.apache.spark.sql.{ DataFrame, SQLContext }
+import org.apache.spark.rdd.{ RDD }
+import org.trustedanalytics.atk.domain.frame.load.{ JdbcArgs }
+import org.trustedanalytics.atk.domain.schema.DataTypes._
+
+/**
+ * Helper class for creating an RDD from jdbc
+ */
+object LoadJdbcImpl extends Serializable {
+
+  /**
+   * Create a data frame from an jdbc compatible database
+   * @param sc default spark context
+   * @param arguments arguments for jdbc connection (including the initial data filtering)
+   */
+  def createDataFrame(sc: SparkContext, arguments: JdbcArgs): DataFrame = {
+    val urlKey = "url"
+    val dbTableKey = "dbtable"
+
+    val sqlContext = new SQLContext(sc)
+    val url = arguments.url.getOrElse(buildUrl())
+    val connectionArgs: Map[String, String] = if (arguments.driverName.isEmpty) {
+      Map(
+        urlKey -> url,
+        dbTableKey-> arguments.tableName)
+    }
+    else {
+      Map(
+        urlKey -> url,
+        dbTableKey -> arguments.tableName,
+        "driver" -> arguments.driverName.get)
+    }
```

Comment:
rather than duplicate map declaration, I'd try something else, maybe something like this (or other Scala funkiness)...

```
val connectionArgs = Map("url" -> arguments.url.getOrElse(buildUrl()), "dbtable" -> arguments.tableName) ++
      (if (arguments.driverName.isDefined) Map("driver" -> arguments.driverName.get) else Map())
```